SilKit/source/services/logging/Logger.cpp

```diff
@@ -118,32 +314,146 @@ Logger::Logger(const std::string& participantName, Config::Logging config)
         {
 #if _WIN32
             auto stdoutSink = std::make_shared<spdlog::sinks::stdout_sink_mt>();
+
 #else
             auto stdoutSink = std::make_shared<spdlog::sinks::ansicolor_stdout_sink_mt>();
 #endif
-            stdoutSink->set_level(log_level);
-            _logger->sinks().emplace_back(std::move(stdoutSink));
+
+            if (sink.format == Config::Sink::Format::Json && sink.type != Config::Sink::Type::Remote)
+            {
+                using spdlog::details::make_unique; // for pre c++14
+                auto formatter = make_unique<spdlog::pattern_formatter>();
+                formatter->add_flag<epoch_formatter_flag>('E').set_pattern("[%E] [%^%l%$] %v");
+                formatter->set_pattern(jsonpattern);
+                stdoutSink->set_formatter(std::move(formatter));
+                stdoutSink->set_level(log_level);
+                _loggerJson->sinks().emplace_back(std::move(stdoutSink));
+            }
+            else if(sink.type != Config::Sink::Type::Remote)
+            {
+                stdoutSink->set_level(log_level);
+                _loggerSimple->sinks().emplace_back(std::move(stdoutSink));
+            }
             break;
         }
         case Config::Sink::Type::File:
         {
-            //
             auto filename = fmt::format("{}_{:%FT%H-%M-%S}.txt", sink.logName, tmBuffer);
```

Comment:
For now the `.txt` extension is fine (imho), but we might want to change that for the `json` formatted files.

Reply:
Regarding the extension we could use `.ndjson` for 'Newline-Delimited JSON' as proposed [here](https://github.com/ndjson/ndjson-spec) or alternatively `.jsonl` as proposed [here](https://jsonlines.org/). From what I can see, both propose essentially the same thing, and neither are 'official'.

It's probably not a good idea to use `.json` since that would imply a single JSON value.

But yeah, for now `.txt` should be fine, since it can be opened in a text editor anyway.

Reply:
.jsonl looks good, since VS Code recognizes it as JSON.

Reply:
@VLukasBecker thats a really good point, I didn't even check that!